ably/rest/auth.py

```diff
@@ -32,64 +30,84 @@ class Method:
     def __init__(self, ably, options):
         self.__ably = ably
         self.__auth_options = options
-
         self.__basic_credentials = None
         self.__token_credentials = None
         self.__auth_params = None
-        self.__token_details = None
-
-        if options.key_value is not None and options.client_id is None:
+        self._token_details = None
+        self.__auth_method = Auth.Method.BASIC
+        useToken = self.shouldUseTokenAuth()
+        if options and options.keyValue is not None:
             # We have the key, no need to authenticate the client
             # default to using basic auth
             log.debug("anonymous, using basic auth")
-            self.__auth_method = Auth.Method.BASIC
-            basic_key = "%s:%s" % (options.key_id, options.key_value)
+            
+            basic_key = "%s:%s" % (options.keyId, options.keyValue)
             basic_key = base64.b64encode(basic_key.encode('utf-8'))
             self.__basic_credentials = basic_key.decode('ascii')
-            return
+        elif not options.auth_token and not useToken:
+            raise AblyException(reason="Neither Token auth nor Basic auth can be setup. Provide a key, or a token, or the means to request a token.",
+                                status_code=400,
+                                code=40000)
+        
+        if useToken:
+            # Using token auth
+            self.__auth_method = Auth.Method.TOKEN
+
+            if options.auth_token:
+                self._token_details = TokenDetails(id=options.auth_token)
+            else:
+                self._token_details = None
+
+            if options.auth_callback:
+                log.debug("using token auth with auth_callback")
```

Comment:
There seems to be a a lot of inconsistent use of underscore case and then camel case.  According to https://www.python.org/dev/peps/pep-0008/, Python uses underscore case, so I think we need to fix the new keyId and keyValue and any other camelCase variables to match the Python style guide.